groups/bsl/bslstl/bslstl_monostate.t.cpp

```diff
@@ -0,0 +1,309 @@
+// bslstl_monostate.t.cpp                                             -*-C++-*-
+#include <bslstl_monostate.h>
+
+#include <bslmf_issame.h>
+
+#include <bsls_asserttest.h>
+#include <bsls_bsltestutil.h>
+
+#include <bslstl_hash.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+using namespace BloombergLP;
+
+// ============================================================================
+//                             TEST PLAN
+// ----------------------------------------------------------------------------
+//                             Overview
+//                             --------
+// The type under test is 'bsl::monostate', a default constructible type whose
+// interface and contract is dictated by the C++ standard.  If 'std::monostate'
+// type is available, we need to check that 'bsl::monostate' is a typedef to
+// the standard's type.  If 'std::monostate' type isn't available, we need to
+// check that 'bsl::monostate' satisfies the interface and contract of
+// 'std::monostate'.
+//
+// ----------------------------------------------------------------------------
+//
+// free functions:
+// [ 3] operator==(monostate, monostate)
+// [ 3] operator!=(monostate, monostate)
+// [ 3] operator<(monostate, monostate)
+// [ 3] operator>(monostate, monostate)
+// [ 3] operator<=(monostate, monostate)
+// [ 3] operator>=(monostate, monostate)
+//
+// [ 5] hashAppend(HASHALG& hashAlg, const monostate&)
```

Comment:
This needs to match exactly the line in the "Testing" section of test case 5.
```suggestion
// [ 5] void hashAppend(HASHALG& hashAlg, const monostate&);
```

Reply:
what's guidance here ? Should i prefer the version with or without the return type ?

Reply:
I *think* we usually put the return type unless it would cause the line to exceed 80 characters, but don't quote me on that. Some of the test driver-specific rules are not documented anywhere in the coding standards and since I'm relatively new on the team, I don't know all the details myself.

No matter what, it always needs to match what's in the test case.

At some point in the project when the component is supposed to be fully polished BDE-style, presumably bde-verify will tell us if there are any style adjustments required.

Reply:
done